polaris-react/src/components/Grid/Grid.scss

```
@@ -0,0 +1,75 @@
+@import '../../styles/common';
+
+.Grid {
+  display: grid;
+  gap: var(--pc-grid-gap-xs, var(--p-space-4));
+  grid-template-columns: repeat(var(--pc-grid-columns-xs, 2), minmax(0, 1fr));
+
+  @media #{$p-breakpoints-sm-up} {
+    gap: var(--pc-grid-gap-sm, var(--p-space-4));
+    grid-template-columns: repeat(var(--pc-grid-columns-sm, 4), minmax(0, 1fr));
+  }
+
+  @media #{$p-breakpoints-md-up} {
+    gap: var(--pc-grid-gap-md, var(--p-space-4));
+    grid-template-columns: repeat(var(--pc-grid-columns-md, 6), minmax(0, 1fr));
+  }
+
+  @media #{$p-breakpoints-lg-up} {
+    gap: var(--pc-grid-gap-lg, var(--p-space-4));
+    grid-template-columns: repeat(
+      var(--pc-grid-columns-lg, 12),
+      minmax(0, 1fr)
+    );
+  }
+
+  @media #{$p-breakpoints-xl-up} {
+    gap: var(--pc-grid-gap-xl, var(--p-space-4));
+    grid-template-columns: repeat(
+      var(--pc-grid-columns-xl, 12),
+      minmax(0, 1fr)
+    );
+  }
+}
+
+.Cell {
+  min-width: 0;
+}
+
+@for $i from 1 through 2 {
+  .grid-#{$i}-column-xs {
+    grid-column-end: span #{$i};
+  }
+}
+
+@for $i from 1 through 4 {
+  @media #{$p-breakpoints-sm-up} {
+    .grid-#{$i}-column-sm {
+      grid-column-end: span #{$i};
+    }
+  }
+}
```

```suggestion
@media #{$p-breakpoints-sm-up} {
  @for $i from 1 through 4 {
    .grid-#{$i}-column-sm {
      grid-column-end: span #{$i};
    }
  }
}
```

**question:** Would this pattern result in cleaner compiled CSS? Kind of looks like we will be generating 36 media queries instead of 5.

---

Good point!